### ISSUE TITLE
Lesson 3 Task 7

### ISSUE THREAD

**Issue by NetSage** [JetBrains/pycharm-courses#5]
I'm not sure what the expected result was on this one but I don't think it's what I'm getting.  I tried multiple things until I could get it to work though.  However the first one to work doesn't seem right as it returned 

```
Traceback (most recent call last):
  File "C:/Users/Alexander/PycharmProjects/PythonIntroduction/lesson3/task7/len_function.py", line 6, in <module>
    first_half = int(phrase[:43])
ValueError: invalid literal for int() with base 10: '\nIt is a really long string\ntriple-quoted s'
```

With the input of 

```
phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
first_half = int(phrase[:43])
print(first_half)
```

And then passes the lesson.

**Comment by silentscreamz411** [comment]
I'm doing it now and I can't get anything to go through, I am new to programming just trying to learn but this one has lost me.

**Comment by NetSage** [comment]
That's why I made this this issue silent.  I'm a noob as well but pretty sure there is something wrong with the check on this task.  Because the input I used to pass the lesson gave an error with python  But, ones that return the expected answer(at least as far as I can tell) fail the task.

**Comment by ktisha** [comment]
Hi NetSage, 

we changed the task description to "Get the first half of the string stored in the variable ‘phrase’" to make it more clear what we are expecting as the answer.
Remember about len() function which is used to count how many characters a string contains so you are not supposed to count 43 yourself. To get this number you could do `int(len(phrase)/2)`. Note: `int()` here is used to convert float 43.0 (returned by `len(phrase)/2`) to integer 43.

In your example:
`phrase[:43]` returns a string. `int(phrase[:43])` throws an exception because string cannot be cast to integer if it doesnt contain integer value.

**Comment by BlindStag** [comment]
this wouldn't complete the task for me until i changed the print function to return, no idea why because i don't think return makes sens unless its part of a defined function? confusing for a beginner like me for sure. bad task.

**Comment by DeadMarshall** [comment]
Hi @ktisha, 
`phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
first_half = type here
print(first_half)`

Using `int(len(phrase)/2)` gives me the output of 44 but the task doesn't get marked as completed. The program reminds me to use string slicing so If I then change the `first_half` variable to = `phrase[0:44]` it will print the first half of the string but it still wont mark as completed.  I believe this is the answer that is being sought but to get it to mark as completed I have to put the following even though it throws 2 errors and 1 warning and the output shows my invalid syntax.

`phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
first_half = (int((len(phrase)/2))phrase[0:44]
print(first_half)`

![image](https://cloud.githubusercontent.com/assets/16566619/17186596/b5abc45e-53ea-11e6-838d-4a3e24c5a0f2.png)

**Comment by CarnegieL** [comment]
Just to point out to those stuck and the above issues, the Task isn't bugged, there is a clean way to solve this. It is looking for half of the text (string). Part of the answer has been provided above by @ktisha

**Comment by VikramBankar** [comment]
Still not working.
Please define what **"Get the first half of the string stored in the variable phrase."** Means
To me it sounds like: Get the string not the length.
What exactly does the output expect?

**Comment by lancelote** [comment]
@VikramBankar You need to extract first half of the string stored in the variable `phrase` (with slicing, `len` and `int` for example) and store it in the `first_half` variable.

In the end `first_half` variable should be:

```

It is a really long string
triple-quoted st
```

**Comment by shake360** [comment]
phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
first_half = int(len(phrase)/2)
print(first_half)


that is what im writing and it still doesnt work. my thinking is as follows. initialize the count. and then divide.

**Comment by lancelote** [comment]
@shake360 That's a correct logic but you forgot to use slicing, for now you store the number 44 in the `first_half` variable, not a first half of the phrase itself as expected.

**Comment by Michael-Wayne** [comment]
@shake360 

phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
first_half = phrase[:int(len(phrase)/2)]
print(first_half)

This is what you should have

**Comment by dkwolfe** [comment]
Good Evening;

I'm playing with the same thing, and I put in exactly what Michael-Wayne said, and I keep getting the error "Too short string in the output", but I having this as an output: "It is a really long string
triple-quoted st"

What am I doing wrong?

**Comment by lancelote** [comment]
@dkwolfe Could you attach a screenshot of your solution please? Not sure where the problem is but Michael-Wayne code works fine for me.

**Comment by Anyaoha** [comment]
phrase[:44]

**Comment by fbaskoro86** [comment]
phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""
**first_half = phrase[0:(int(len(phrase)/2))]**
print(first_half)

-----------
i just tried like that, and passed the check task

**Comment by Hamberfim** [comment]
This definitely needs "word-smithed" i.e., re-word even further to be more clear. Furthermore, the previous lessons need to cover a bit more on acceptable syntax for slicing. Like everyone else that struggled with this one I was able to get either the number or chars. in the string using len() and divide it or I was able to get the first half of the "paragraph", "words listed"...phrase into the print function but I didn't make the connection to make it a single statesmen within the slice brackets. There wasn't any indication that anything other numeric values could be used within the brackets.


Here is how I got there:
phrase = """
It is a really long string
triple-quoted strings are used
to define multi-line strings
"""

phraseLen = len(phrase)
print(int(phraseLen / 2))
first_half = phrase[:44]
print(first_half)
